src/kernel.rs

```diff
@@ -0,0 +1,171 @@
+use crate::{println_with_prefix, yield_into};
+use anyhow::{anyhow, Result};
+use semver::Version;
+use std::{fmt, fs, io::Write, path::Path};
+
+const REL_INST_PATH: &str = "EFI/aosc/";
+const SRC_PATH: &str = "/boot/";
+const UCODE_PATH: &str = "/boot/intel-ucode.img";
+
+/// A kernel struct for parsing kernel filenames
+pub struct Kernel {
+    pub version: Version,
+    pub distro: String,
+    pub flavor: String,
+}
+
+impl Ord for Kernel {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.version.cmp(&other.version)
+    }
+}
+
+impl PartialOrd for Kernel {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq for Kernel {
```

Comment:
I believe `PartialEq` and `Eq` can be automatically implemented using derive macro?